functions/paperChannelTenders/src/middlewares/validators.ts

```diff
@@ -1,8 +1,39 @@
-import validatorMiddleware from '@middy/validator';
-import { transpileSchema } from '@middy/validator/transpile';
-import { schemaEventGetTenders } from '../types/schema-request-types';
+import {
+  CostEventSchema,
+  CostsEventSchema, DeliveryDriversEventSchema,
+  Event,
+  EventSchema,
+  TenderActiveEventSchema,
+  TendersEventSchema,
+} from '../types/schema-request-types';
 
+const validatorEvent = (event: unknown): Event => {
+  const parsedEvent = EventSchema.safeParse(event);
 
-export const validatorMiddlewareGetTenders = validatorMiddleware({
-  eventSchema: transpileSchema(schemaEventGetTenders)
-})
\ No newline at end of file
+  if (!parsedEvent.success) {
+    //console.error("Invalid event data:", parsedEvent.error);
+    throw new Error("Invalid event data");
+  }
+
+  if (TendersEventSchema.safeParse(parsedEvent.data).success) {
```

Comment:
Toglierei la ridondanza nel codice 

  const eventSchemas = [
    TendersEventSchema,
    TenderActiveEventSchema,
    CostsEventSchema,
    CostEventSchema,
    DeliveryDriversEventSchema,
  ];

  for (const schema of eventSchemas) {
    const result = schema.safeParse(parsedEvent.data);
    if (result.success) {
      return result.data;
    }
  }

  throw new Error("Unknown event type");

Reply:
resolved